Modules/mmapmodule.c

```diff
@@ -307,8 +318,23 @@ mmap_read_method(mmap_object *self,
     remaining = (self->pos < self->size) ? self->size - self->pos : 0;
     if (num_bytes < 0 || num_bytes > remaining)
         num_bytes = remaining;
+
+#if defined(MS_WIN32) && !defined(DONT_USE_SEH)
+    EXCEPTION_RECORD record;
+    __try {
+        result = PyBytes_FromStringAndSize(&self->data[self->pos], num_bytes);
+        self->pos += num_bytes;
+    }
+    __except (HandlePageException(GetExceptionInformation(), &record)) {
+        NTSTATUS code = record.ExceptionInformation[2];
```

Comment:
`OSError` doesn't support `NTSTATUS` codes. You'll have to dynamically call `ULONG RtlNtStatusToDosError(NTSTATUS Status)` from the system library "ntdll.dll".

Reply:
Or we can raise a generic `OSError` and format the error code into the string. It's incredibly unlikely that the caller can actually handle any specific errors, so all that is likely to happen is logging and cleanup. We don't need to get winerror perfect for those.